dali/test/python/test_dali_variable_batch_size.py

```diff
@@ -372,18 +445,294 @@ def pipe_no_input(max_batch_size, input_data, device):
+        pipe = Pipeline(batch_size=max_batch_size, num_threads=4, device_id=0)
+        with pipe:
+            # just to drive the variable batch size.
+            data = fn.external_source(source=input_data, cycle=False, device=device, layout=input_layout)
+            out = fn.expand_dims(data, axes=[0, 2], new_axis_names="YZ")
+            out = fn.squeeze(out, axis_names="Z")
+        pipe.set_outputs(out)
+        return pipe
+
+    check_pipeline(generate_data(31, 13, image_like_shape_generator, lo=0, hi=255, dtype=np.uint8),
+                   pipeline_fn=pipe, input_layout="HWC")
+
+
+def test_box_encoder_op():
+    def coco_anchors():
+        anchors = []
+
+        fig_size = 300
+        feat_sizes = [38, 19, 10, 5, 3, 1]
+        feat_count = len(feat_sizes)
+        steps = [8., 16., 32., 64., 100., 300.]
+        scales = [21., 45., 99., 153., 207., 261., 315.]
+        aspect_ratios = [[2], [2, 3], [2, 3], [2, 3], [2], [2]]
+
+        fks = []
+        for step in steps:
+            fks.append(fig_size / step)
+
+        anchor_idx = 0
+        for idx in range(feat_count):
+            sk1 = scales[idx] / fig_size
+            sk2 = scales[idx + 1] / fig_size
+            sk3 = sqrt(sk1 * sk2)
+
+            all_sizes = [[sk1, sk1], [sk3, sk3]]
+
+            for alpha in aspect_ratios[idx]:
+                w = sk1 * sqrt(alpha)
+                h = sk1 / sqrt(alpha)
+                all_sizes.append([w, h])
+                all_sizes.append([h, w])
+
+            for sizes in all_sizes:
+                w, h = sizes[0], sizes[1]
+
+                for i in range(feat_sizes[idx]):
+                    for j in range(feat_sizes[idx]):
+                        cx = (j + 0.5) / fks[idx]
+                        cy = (i + 0.5) / fks[idx]
+
+                        cx = max(min(cx, 1.), 0.)
+                        cy = max(min(cy, 1.), 0.)
+                        w = max(min(w, 1.), 0.)
+                        h = max(min(h, 1.), 0.)
+
+                        anchors.append(cx - 0.5 * w)
+                        anchors.append(cy - 0.5 * h)
+                        anchors.append(cx + 0.5 * w)
+                        anchors.append(cy + 0.5 * h)
+
+                        anchor_idx = anchor_idx + 1
+        return anchors
+
+    def pipe(max_batch_size, input_data, device, input_layout=None):
+        pipe = Pipeline(batch_size=max_batch_size, num_threads=4, device_id=0)
+        with pipe:
+            boxes, lables = fn.external_source(device=device, source=input_data, num_outputs=2)
+            processed, _ = fn.box_encoder(boxes, lables, anchors=coco_anchors())
+        pipe.set_outputs(processed)
+        return pipe
+
+    def get_data(batch_size):
+        obj_num = random.randint(1, 20)
+        test_box_shape = [obj_num, 4]
+        test_lables_shape = [obj_num, 1]
+        bboxes = [(np.random.randint(0, 255, size=test_box_shape, dtype=np.uint8) / 255).astype(dtype=np.float32) for _ in range(batch_size)]
```

Comment:
Nitpick:
```suggestion
        bboxes = [np.random.random(size=test_box_shape, dtype=np.uint8) .astype(dtype=np.float32) for _ in range(batch_size)]
```
Just floats in [0, 1]?

And a question: does it work ok with invalid boxes?

Reply:
Done
No idea, but I don't think this is the place to test this aspect of bbox encoder.